client/my-sites/themes/themes-magic-search-card/index.jsx

```diff
@@ -71,15 +71,20 @@ const ThemesMagicSearchCard = React.createClass( {
 
 		return (
 			tokens.map( ( token, i ) => {
-				let classname = 'themes-magic-search-card__search-text';
-
 				if ( token.trim() === '' ) {
-					classname = 'themes-magic-search-card__search-white-space';
+					return <span className="themes-magic-search-card__search-white-space" key={ i }>{ token }</span>; // use shortid for key
 				} else if ( filterIsValid( token ) ) {
-					classname = 'themes-magic-search-card__search-token';
+					const separator = ':';
+					const [ taxonomy, filter ] = token.split( separator );
```

Comment:
@seear Should we be using `../theme-filters` `getTaxonomy` and `getTerm` instead?